File: src/Service/Service.Core/Infrastructure/PipeCommunicationResolver.cs
using Service.Core.Abstractions.Interfaces;
using System;
using System.Collections.Generic;
using System.IO;
using System.IO.Pipes;
using System.Text;

namespace Service.Core.Infrastructure
{
    internal class PipeCommunicationResolver : IServiceCommunicationResolver
    {
        public event Func<byte[], byte[]> OnCommunicationCreated;
        
        private NamedPipeServerStream pipe;
        private string pipeName;

        public PipeCommunicationResolver(string pipeName)
        {
            this.pipeName = pipeName;
        }

        public void Listen()
        {
            while (true) {
                pipe = new NamedPipeServerStream(pipeName, PipeDirection.InOut);

                pipe.BeginWaitForConnection(OnConnected, null);
            }
        }


        #region Private

        private void OnConnected(IAsyncResult result)
        {
            pipe.EndWaitForConnection(result);

            using (var memoryStream = new MemoryStream())
            {
                pipe.CopyTo(memoryStream);
                
            }
        }
 

        #endregion
    }
}

File: src/Library/sources/PipeCommunicationResolver.cpp
#include "../include/PipeCommunicationResolver.h"

bool Infrastructure::PipeCommunicationResolver::InitialiseCommunication()
{
	return false;
}

Abstractions::Bytes Infrastructure::PipeCommunicationResolver::ExecuteRequest(const Abstractions::Bytes& bytes)
{
	Abstractions::Bytes result(nullptr, 0);

	return result;
}

bool Infrastructure::PipeCommunicationResolver::FinaliseCurrentCommunication()
{
	return false;
}
